### PR TITLE
Allow running local binaries

### PR DESCRIPTION
**Pull request by ehmicky**
Mentioned in #14.

This PR allows running locally installed binaries.

### CODE REVIEW

**Comment by sindresorhus** [comment]
Should it maybe be opt-in?

**Comment by ehmicky** [comment]
Hum, I actually always wondered why it was opt-in in Execa?

The cons I see are rather minor. Performance-wise, the OS will do `stat` calls through more directories in the `PATH`. However, on my machine, this is <=1ms (depending on how much the OS parallelizes and caches those calls), which is nothing compared to the time need to allocate resources for the new process, then to load the process runtime.

Also, it passes the full `process.env` as environment variables to the subprocess, instead of specifying nothing and inheriting it. This probably has a performance impact, but of an even smaller magnitude.

On my machine, I have had `node_modules/.bin` in my `PATH` (from my `.bashrc`) and have never encountered a practical issue.

The main pro is that executing binaries from your `package.json` dependencies is a very common use case. Users might assume to work out-of-the-box.

Also, not having an option would keep the API smaller.

What are your thoughts on this?

**Comment by sindresorhus** [comment]
See https://github.com/sindresorhus/execa/pull/314 for context.

Preferring local binaries for user scripts definitely makes sense, but for reusable packages, it's not that commonly used. And it may even cause problems if you want to execute a global binary, and then a different version exists locally in `node_modules`, which can cause unpredictability. And because of hoisting and dependency resolution, it may not even reproduce every time.

I don't feel strongly about it though and could be convinced otherwise.

**Comment by sindresorhus** [comment]
> Should we expose this under a boolean option, defaulting to false?

👍 

> If so, how to call it? preferLocal or something else?

Thought about it and couldn't come up with a better name, so yeah `preferLocal`, unless you have a better suggestion.

**Comment by ehmicky** [comment]
Done. :+1:

File: test.js [hunk 1]
```diff
@@ -737,52 +737,109 @@ test('Handles non-existing command, shell', async t => {
...
+test('options.preferLocal true can pass arguments to local npm binaries, `', testLocalBinary, '`');
+test('options.preferLocal true can pass arguments to local npm binaries, <', testLocalBinary, '<');
+test('options.preferLocal true can pass arguments to local npm binaries, >', testLocalBinary, '>');
+test('options.preferLocal true can pass arguments to local npm binaries, &', testLocalBinary, '&');
+test('options.preferLocal true can pass arguments to local npm binaries, |', testLocalBinary, '|');
+test('options.preferLocal true can pass arguments to local npm binaries, ;', testLocalBinary, ';');
+test('options.preferLocal true can pass arguments to local npm binaries, ,', testLocalBinary, ',');
+test('options.preferLocal true can pass arguments to local npm binaries, space', testLocalBinary, ' ');
+test('options.preferLocal true can pass arguments to local npm binaries, *', testLocalBinary, '*');
+test('options.preferLocal true can pass arguments to local npm binaries, ?', testLocalBinary, '?');
```
Comment (sindresorhus) [inline test.js:842]: Doesn't have to be this PR, but maybe pass in the repeatable part of the titles here as a variable to reduce duplication.

**Pull request merged**